magma/primitives/register.py

```diff
@@ -206,3 +206,13 @@ def __init__(self, T: Kind = None,
         elif (has_reset or has_resetn):
             I = Mux(2, T)()(I, init, reset_select)
         reg.I @= as_bits(I)
+
+
+def register(value, **kwargs):
+    T = type(value).qualify(Direction.Undirected)
```

Comment:
@leonardt one question is if the un-qualification logic should be in the base Register generator or not. not a big deal either way

Reply:
Hmm, I'm not sure.  I think normally a user would say the type explicitly, e.g. `Register(Array[N, Bits[M])`, so it's undirected.  In this case, when you call `register` it's often directed (unless it's an intermediate value) so it makes sense to call it here.  I think the one case I can imagine is if a user says something like `Register(type(v))` where `v` is a directed value.  Often though, in this case, they could just call `register(v)`, but maybe for somer reason they want to just create a register of the same type without using the value as the input, so maybe it's useful to have in the base generator even if it's an obscure case.  We could use a decorator that handles this conversion since it's probably a similar issue in other generators.

Reply:
Makes sense -- agree that its more specific to `register` than `Register` because the common case is an explicit type passed in rather than `type(value)`